src/vmm_core.mli

```diff
@@ -22,48 +22,67 @@ module IM : sig
 end
 
 module Name : sig
-  type t
+  module Label : sig
+    type t
+
+    val compare : t -> t -> int
+    val equal : t -> t -> bool
+    val is_empty : t -> bool
+    val empty : t
+
+    val of_string : string -> (t, [> `Msg of string ]) result
+    val to_string : t -> string
+  end
+
+  module Path : sig
+    type t
+
+    val compare : t -> t -> int
+    val equal : t -> t -> bool
+
+    val of_label : Label.t -> t
+    val of_labels : Label.t list -> t
+
+    val to_string : t -> string
+    val of_string : string -> (t, [> `Msg of string ]) result
+
+    val to_labels : t -> Label.t list
+
+    val root : t
+    val is_root : t -> bool
+    val parent : t -> t
 
-  type path
+    val append : t -> string -> (t, [> `Msg of string ]) result
+    val append_exn : t -> string -> t
```

Comment:
not sure whether we need this anymore, since you've Label.of_string_exn and append_label

Reply:
Yes, I would drop both `append` and `append_exn` as it simplifies the `Path` code. We can consider adding `Label.of_string_exn` though.